source/Halibut/Diagnostics/HalibutLimits.cs

```diff
@@ -107,4 +108,102 @@ public static TimeSpan SafeTcpClientPooledConnectionTimeout
+        /// </remarks>
+        public int RewindableBufferStreamSize { get; set; } =  HalibutLimits.RewindableBufferStreamSize;
+        
+        /// <summary>
+        /// Amount of time to wait for a TCP or SslStream write to complete successfully
+        /// </summary>
+        public TimeSpan TcpClientSendTimeout { get; set; } =  HalibutLimits.TcpClientSendTimeout;
+        
+        /// <summary>
+        /// Amount of time to wait for a TCP or SslStream read to complete successfully
+        /// </summary>
+        public TimeSpan TcpClientReceiveTimeout { get; set; } =  HalibutLimits.TcpClientReceiveTimeout;
+        
+        /// <summary>
+        /// Amount of time a connection can stay in the pool
+        /// </summary>
+        public TimeSpan TcpClientPooledConnectionTimeout { get; set; } =  HalibutLimits.TcpClientPooledConnectionTimeout;
+        
+        public TimeSpan TcpClientHeartbeatSendTimeout { get; set; } =  HalibutLimits.TcpClientHeartbeatSendTimeout;
+        public TimeSpan TcpClientHeartbeatReceiveTimeout { get; set; } =  HalibutLimits.TcpClientHeartbeatReceiveTimeout;
+        
+        /// <summary>
+        /// Amount of time to wait for a successful TCP or WSS connection
+        /// </summary>
+        public TimeSpan TcpClientConnectTimeout { get; set; } =  HalibutLimits.TcpClientConnectTimeout;
+        
+        /// <summary>
+        /// The amount of time client will wait for a message to be added to the polling request queue
+        /// before returning a null response to the server. This does not generate an error and the server would immediate re-request.
+        /// </summary>
+        public TimeSpan PollingQueueWaitTimeout { get; set; } =  HalibutLimits.PollingQueueWaitTimeout;
+#pragma warning restore CS0612
+        // After a client/server message exchange is complete, the client returns
+        // the connection to the pool but the server continues to block and reads
+        // from the connection until the TcpClientReceiveTimeout.
+        // If TcpClientPooledConnectionTimeout is greater than TcpClientReceiveTimeout
+        // when the client goes to the pool to get a connection for the next
+        // exchange it can get one that has timed out, so make sure our pool
+        // timeout is smaller than the tcp timeout.
+        public TimeSpan SafeTcpClientPooledConnectionTimeout
+        {
+            get
+            {
+                if (TcpClientPooledConnectionTimeout < TcpClientReceiveTimeout)
+                {
+                    return TcpClientPooledConnectionTimeout;
+                }
+                else
+                {
+                    var timeout = TcpClientReceiveTimeout - TimeSpan.FromSeconds(10);
+                    return timeout > TimeSpan.Zero ? timeout : TcpClientReceiveTimeout;
+                }
+            }
+        }
+
+        public HalibutTimeoutsAndLimits WithTcpClientReceiveTimeout(TimeSpan tcpClientReceiveTimeout)
```

Comment:
Why is this one special? All the properties have public set methods so it this needed? Didi you mean to remove the set methods on properties.

Would this be better suited as an extension method if your trying to create a bit of a fluent API for setting values?

Reply:
Yeah I guess it can be an extension method.